### DISCUSSION TITLE
AttributeError: 'GCNConv' object has no attribute 'aggr_module'

### DISCUSSION THREAD

**Discussion by seriouslym** [pyg-team/pytorch_geometric#6293] [Q&A]
In May of this year, I used pyg (torch-1.10.0+cu113) to train several models (GCN, PPNP, GAT) and saved them. When I loaded the models in the same way today, some errors were reported.
1: TypeError: edge_update() missing 6 required positional arguments: 'alpha_j', 'alpha_i', 'edge_attr', 'index', 'ptr', and 'size_i'
2: AttributeError: 'GCNConv' object has no attribute 'aggr_module'

**Comment by rusty1s** [comment]
In general, we cannot assume backward compatibility of saving and loading models across different PyG versions since this wouldn't allow us to make any changes to the underlying layers. At best, you should always store only the `state_dict` of a model rather than the whole object. If you want to load your models, you have to jump back to the PyG version that was used to store the model.
